ssts/segmentation/persistence_watershed_algorithm.py

```diff
@@ -141,8 +141,11 @@ def merge_tree(maxima, edges):
             if values[vc] < values[uc]:  # if later label has lower max
                 uc, vc = vc, uc
 
-            # source, target, persistence (size of lower label hill)
-            pairs.append((uc, vc, values[uc] - val))
-            components[uc] = components[vc]  # lower label maps to higher label
+            pairs.append((uc, vc, values[uc] - val)) # source, target, persistence (size of lower label hill)
+
+            # NOTE without this constraint, artifacts appear in the output plot
+            if (values[uc] - val) < pers_thresh:
```

Comment:
@ponl This is where the artifacts were coming from. We should talk about this because I don't think this is optimal